### PR TITLE
TLS Improvments

### PR DESCRIPTION
**Pull request by jtg-dbs**
Generate TLS Certificate

The Nginx generates a TLS Certificate on every start if there is the environment variable is not set. You can add your own cetrificate by mounting it to /etc/nginx/ssl/ and setting GENERATE_TLS_CERTIFICATE to false. You can activate the TLS configuration fo nginx with the environment variable USE_TLS.

From my point of view it is an security issue that the certificate is inlcuded during the docker build step. It should be added at the runtime not at buildtime.

Hardening on the TLS Configuration of nginx.

On behalf of DB Systel GmbH

----

**Note: DefectDojo is now on Python3 and Django 2.2.1 Please submit your pull requests to the 'dev' branch as the 'legacy-python2.7' branch is only for bug fixes. Any new features submitted to the legacy branch will be ignored and closed.**

When submitting a pull request, please make sure you have completed the following checklist:

- [ x] Your code is flake8 compliant 
- [ x] Your code is python 3.5 compliant
- [ x] If this is a new feature and not a bug fix, you've included the proper documentation in the ReadTheDocs documentation folder. https://github.com/DefectDojo/Documentation/tree/master/docs or provide feature documentation in the PR.
- [ x] Model changes must include the necessary migrations in the dojo/dd_migrations folder.
- [ ] Add applicable tests to the unit tests.

### CODE REVIEW

**Comment by NRGLine4Sec** [comment]
Hello, Why you don't add TLS 1.3 ?

**Comment by jtg-dbs** [comment]
added TLS1.3

**Comment by ptrovatelli** [comment]
hello @jtg-dbs ; 

getting the certificate into the container at build time was just an easy way to deploy a custom certificate to the container with an already known command: that of building the nginx image + an override (see docker-compose.override.https.yml). I agree it may be confusing though when several instances must use the same image with a different certificate, altough it's always possible to change the certificate after build time. 
 Could you update DOCKER.MD please? that part: `Run with docker using https` with the commands that need to be run in order to place a custom certificate into the container and see what will happen to docker-compose.override.https.yml with your solution? for example your changes in docker-compose.yml are already there (just with a different port).
also, what list of ciphers is that? can you maybe drop a comment about it in the code? 
thanks!

**Comment by jtg-dbs** [comment]
@ptrovatelli  
Thank you for the comment. i updated the MD and inserted the link to the mozilla ssl generator which i used.

File: Dockerfile.nginx [hunk 1]
```diff
@@ -57,17 +57,23 @@ RUN \
...
-  if [ -f /*.key -o -f /*.cer ]; then chown 1001 /*.key /*.cer; fi && \
+  mkdir -p /etc/nginx/ssl && \
+  chown 1001 /etc/nginx/ssl/ &&\
   true
 ENV \
   DD_UWSGI_PASS="uwsgi_server" \
   DD_UWSGI_HOST="uwsgi" \
-  DD_UWSGI_PORT="3031"
+  DD_UWSGI_PORT="3031" \
+  GENERATE_TLS_CERTIFICATE="true" \
```
Comment (ptrovatelli) [inline Dockerfile.nginx]: shouldn't that be "True" / "False"? (capital T/F)

File: DOCKER.md [hunk 2]
```diff
@@ -218,6 +219,15 @@ chmod 400 nginx/*.key
 
 ```docker build  -t defectdojo/defectdojo-nginx -f Dockerfile.nginx .```
 
+* Run defectDojo with: 
+```
+rm -f docker-compose.override.yml
+ln -s docker-compose.override.https.yml docker-compose.override.yml
+docker-compose up
```
Comment (ptrovatelli) [inline DOCKER.md]: this appears twice. can you put it once only please? maybe create two sections with the pre-requisites for: 
- using a self-signed certificate
- using your own certificate (where to copy the certificate, set generate_certificate=False)

Then describe the common part to both (start with the https override)

**Comment by ptrovatelli** [comment]
other than that it's fine; thanks and sorry for the delay :)

**Comment by jtg-dbs** [comment]
thanks @ptrovatelli for the comments.

**Pull request merged**